Assets/Mirage/Components/NetworkManagerGUI.cs

```diff
@@ -1,18 +1,70 @@
+using System;
+using Mirage.Logging;
 using UnityEngine;
 
 namespace Mirage
 {
     public class NetworkManagerGUI : MonoBehaviour
     {
-        public NetworkManager NetworkManager;
+        private static readonly ILogger logger = LogFactory.GetLogger(typeof(NetworkManagerGUI));
+
+        [Header("Defaults")]
+        [Tooltip("This is the default string displayed in the client address text box.")]
         public string NetworkAddress = "localhost";
 
-        [Range(0.01f, 10f)]
+        [Header("References")]
+        [Tooltip("If enabled, we'll automatically reference NetworkManager in script's initialization.\nIf you've already set the NetworkManager field, auto configuring will not be performed.")]
+        public bool AutoConfigureNetworkManager = true;
+
+        [Tooltip("The value in this property field (if set) will not be overwritten if Auto Configuration is enabled.")]
+        public NetworkManager NetworkManager;
+
+        [Header("Cosmetics")]
+        [Range(0.01f, 10f), Tooltip("Adjusts scale of the GUI elements. 1x is normal size. 2x might be sane for big screens. <= 0.5x or > 2x is silly.")]
         public float Scale = 1f;
+
+        [Tooltip("Select where you want the NetworkManagerGUI elements to be located on the screen.")]
         public TextAnchor GUIAnchor = TextAnchor.UpperLeft;
 
+        private void Awake()
+        {
+            // Coburn, 2023-01-29: 
+            // If automatic configuration of NetworkManager is enabled, then attempt to grab the
+            // NetworkManager that this script is attached to.
+            if (AutoConfigureNetworkManager)
```

Comment:
if it hard to follow this chain of if/else,, would be better to do early returns instead